File: copilot-scraper/api/python/request.py
"""Copilot answer via the Scrapeless Scraper API (scraper.copilot).

Run:
    export SCRAPELESS_API_KEY=your_api_token_here
    python request.py
"""
import os
import json
import requests

ENDPOINT = "https://api.scrapeless.com/api/v2/scraper/execute"


def ask_copilot(prompt: str, country: str = "US", mode: str = "smart") -> dict:
    resp = requests.post(
        ENDPOINT,
        headers={
            "Content-Type": "application/json",
            "x-api-token": os.environ["SCRAPELESS_API_KEY"],
        },
        json={
            "actor": "scraper.copilot",
            "input": {"prompt": prompt, "country": country, "mode": mode},
        },
        timeout=180,
    )
    resp.raise_for_status()
    return resp.json()


if __name__ == "__main__":
    data = ask_copilot("What are the best web scraping tools?")
    # `task_result` holds the answer (result_text, citations, links, ...).
    print(json.dumps(data.get("task_result", {}), indent=2, ensure_ascii=False))

File: copilot-scraper/api/nodejs/request.mjs
// Copilot answer via the Scrapeless Scraper API (scraper.copilot).
//   export SCRAPELESS_API_KEY=your_api_token_here
//   node request.mjs
const ENDPOINT = "https://api.scrapeless.com/api/v2/scraper/execute";

export async function askCopilot(prompt, country = "US", mode = "smart") {
  const resp = await fetch(ENDPOINT, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      "x-api-token": process.env.SCRAPELESS_API_KEY,
    },
    body: JSON.stringify({ actor: "scraper.copilot", input: { prompt, country, mode } }),
  });
  if (!resp.ok) throw new Error(`HTTP ${resp.status}: ${await resp.text()}`);
  return resp.json();
}

const data = await askCopilot("What are the best web scraping tools?");
// `task_result` holds the answer (result_text, citations, links, ...).
console.log(JSON.stringify(data.task_result ?? {}, null, 2));
